lib/src/event.dart

```diff
@@ -1001,37 +1001,50 @@ class Event extends MatrixEvent {
       content['formatted_body'] is String;
 
   // regexes to fetch the number of emotes, including emoji, and if the message consists of only those
-  // to match an emoji we can use the following regex:
-  // (?:\x{00a9}|\x{00ae}|[\x{2600}-\x{27bf}]|[\x{2b00}-\x{2bff}]|\x{d83c}[\x{d000}-\x{dfff}]|\x{d83d}[\x{d000}-\x{dfff}]|\x{d83e}[\x{d000}-\x{dfff}])[\x{fe00}-\x{fe0f}]?
-  // we need to replace \x{0000} with \u0000, the comment is left in the other format to be able to paste into regex101.com
+  // to match an emoji we can use the following regularly updated regex : https://stackoverflow.com/a/67705964
   // to see if there is a custom emote, we use the following regex: <img[^>]+data-mx-(?:emote|emoticon)(?==|>|\s)[^>]*>
-  // now we combind the two to have four regexes:
+  // now we combined the two to have four regexes and one helper:
+  // 0. the pure unicode sequence from the link above
   // 1. are there only emoji, or whitespace
   // 2. are there only emoji, emotes, or whitespace
   // 3. count number of emoji
   // 4- count number of emoji or emotes
+  static const _unicodeSequences =
+      r'\u00a9|\u00ae|[\u2000-\u3300]|\ud83c[\ud000-\udfff]|\ud83d[\ud000-\udfff]|\ud83e[\ud000-\udfff]';
+
   static final RegExp _onlyEmojiRegex = RegExp(
-      r'^((?:\u00a9|\u00ae|[\u2600-\u27bf]|[\u2b00-\u2bff]|\ud83c[\ud000-\udfff]|\ud83d[\ud000-\udfff]|\ud83e[\ud000-\udfff])[\ufe00-\ufe0f]?|\s)*$',
-      caseSensitive: false,
-      multiLine: false);
+    r'^((?:\x{00a9}|\x{00ae}|' + _unicodeSequences + r')[\ufe00-\ufe0f]?|\s)*$',
+    caseSensitive: false,
+    multiLine: false,
+  );
   static final RegExp _onlyEmojiEmoteRegex = RegExp(
-      r'^((?:\u00a9|\u00ae|[\u2600-\u27bf]|[\u2b00-\u2bff]|\ud83c[\ud000-\udfff]|\ud83d[\ud000-\udfff]|\ud83e[\ud000-\udfff])[\ufe00-\ufe0f]?|<img[^>]+data-mx-(?:emote|emoticon)(?==|>|\s)[^>]*>|\s)*$',
-      caseSensitive: false,
-      multiLine: false);
+    r'^((?:\x{00a9}|\x{00ae}|' +
+        _unicodeSequences +
+        r')[\ufe00-\ufe0f]?|<img[^>]+data-mx-(?:emote|emoticon)(?==|>|\s)[^>]*>|\s)*$',
+    caseSensitive: false,
+    multiLine: false,
+  );
   static final RegExp _countEmojiRegex = RegExp(
-      r'((?:\u00a9|\u00ae|[\u2600-\u27bf]|[\u2b00-\u2bff]|\ud83c[\ud000-\udfff]|\ud83d[\ud000-\udfff]|\ud83e[\ud000-\udfff])[\ufe00-\ufe0f]?)',
-      caseSensitive: false,
-      multiLine: false);
+    r'((?:\x{00a9}|\x{00ae}|' +
+        _unicodeSequences +
+        r')[\ufe00-\ufe0f]?)',
+    caseSensitive: false,
+    multiLine: false,
+  );
   static final RegExp _countEmojiEmoteRegex = RegExp(
-      r'((?:\u00a9|\u00ae|[\u2600-\u27bf]|[\u2b00-\u2bff]|\ud83c[\ud000-\udfff]|\ud83d[\ud000-\udfff]|\ud83e[\ud000-\udfff])[\ufe00-\ufe0f]?|<img[^>]+data-mx-(?:emote|emoticon)(?==|>|\s)[^>]*>)',
-      caseSensitive: false,
-      multiLine: false);
+    r'((?:\x{00a9}|\x{00ae}|' +
```

Comment:
I don't think there is any reason to switch these to x instead of u, right? Could you change them back for consistency? Otherwise this looks good to me, thanks!